python/az/aro/azext_aro/_rbac.py

```diff
@@ -48,5 +46,52 @@ def assign_contributor_to_vnet(cli_ctx, vnet, object_id):
     ))
 
 
+def assign_contributor_to_routetable(cli_ctx, master_subnet, worker_subnet, object_id):
+    auth_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_AUTHORIZATION)
+    client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK)
+
+    role_definition_id = resource_id(
+        subscription=get_subscription_id(cli_ctx),
+        namespace='Microsoft.Authorization',
+        type='roleDefinitions',
+        name=DEVELOPMENT_CONTRIBUTOR if rp_mode_development() else CONTRIBUTOR,
+    )
+
+    route_tables = {}
+    for sn in [master_subnet, worker_subnet]:
+        sid = parse_resource_id(sn)
+        subnet = client.subnets.get(resource_group_name=sid['resource_group'],
+                                    virtual_network_name=sid['name'],
+                                    subnet_name=sid['resource_name'])
+        if subnet.route_table is not None and \
+           not has_assignment(auth_client.role_assignments.list_for_scope(subnet.route_table.id),
+                              role_definition_id, object_id):
```

Comment:
can you bring this check down under `for rt in route_tables` ?  That way if there are 2 subnets linked to the same route table, you'll do the has_assignment check + associated API call once rather than twice.

Reply:
I'd propose checking if the route table id exists in the dict and if it does, skip.  This will provide the same functionality as the `set()` suggestion.  WDYT?

Reply:
No, it's not the same.  This code will still do multiple has_assignment calls if both subnets refer to the same route table and the assignments are present in both cases.

Reply:
Maybe I misunderstand.  I added this line https://github.com/Azure/ARO-RP/pull/728/files#diff-eca84667637a895b084e0c417ca734ecR67 which should short circuit if the routeid is already in the set.  If this is the case then the has_assignments call will not be made.

Was this a misunderstanding on my part?

Reply:
After discussion, I have updated the PR to match @jim-minter's suggested change.